sklearn/datasets/openml.py

```diff
@@ -0,0 +1,241 @@
+        data_found = True
+        try:
+            json_string = urlopen(_SEARCH_NAME.format(name) +
+                                  "/data_version/{}/status/deactivated".format(
+                                      version))
+        except HTTPError as error:
+            if error.code == 412:
+                data_found = False
+            else:
+                raise error
+
+    if not data_found:
+        # not in except for nicer traceback
+        if version == "active":
+            raise ValueError("No active dataset {} found.".format(name))
+        raise ValueError("Dataset {} with version {}"
+                         " not found.".format(name, version))
+
+    json_data = json.loads(json_string.read().decode("utf-8"))
+    return json_data['data']['dataset'][0]
+
+
+def _get_data_description_by_id(data_id):
+    data_found = True
+    try:
+        json_string = urlopen(_DATA_INFO.format(data_id))
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+    if not data_found:
+        # not in except for nicer traceback
+        raise ValueError("Dataset with id {} "
+                         "not found.".format(data_id))
+    json_data = json.loads(json_string.read().decode("utf-8"))
+    return json_data['data_set_description']
+
+
+def _get_data_features(data_id):
+    data_found = True
+    try:
+        json_string = urlopen(_DATA_FEATURES.format(data_id))
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+    if not data_found:
+        # not in except for nicer traceback
+        raise ValueError("Dataset with id {} "
+                         "not found.".format(data_id))
+    json_data = json.loads(json_string.read().decode("utf-8"))
+    return json_data['data_features']['feature']
+
+
+def _download_data(url):
+    response = urlopen(url)
+    if sys.version_info[0] == 2:
+        # Python2.7 numpy can't handle unicode?
+        arff = loads(response.read())
+    else:
+        arff = loads(response.read().decode('utf-8'))
+
+    response.close()
+    return arff
+
+
+def _convert_numericals(data, name_feature):
+    for feature in name_feature.values():
+        if feature['data_type'] == "numeric":
+            idx = int(feature['index'])
+            data[:, idx] = data[:, idx].astype(np.float)
+    return data
+
+
+def fetch_openml(name_or_id=None, version='active', data_home=None,
+                 target_column_name='default-target', memory=True):
+    """Fetch dataset from openml by name or dataset id.
+
+    Datasets are uniquely identified by either an integer ID or by a
+    combination of name and version (i.e. there might be multiple
+    versions of the 'iris' dataset).
+
+    Parameters
+    ----------
+    name_or_id : string or integer
+        Identifier of the dataset. If integer, assumed to be the id of the
+        dataset on OpenML, if string, assumed to be the name of the dataset.
```

Comment:
I though we were against radical change of behaviour based on type (can't remember where this was discussed exactly). Why not make 2 parameters `name` and `_id` or `uid`?

Reply:
I agree. The Javatonic way would be to provide two functions, one of obtaining by name / version, one of obtaining by id (where one calls to the other)

@rth @amueller what is your take on this?